haystack/nodes/prompt/prompt_node.py

```diff
@@ -562,6 +552,9 @@ def __init__(
         devices: Optional[List[Union[str, torch.device]]] = None,
     ):
         super().__init__()
+        if prompt_templates is None:
+            prompt_templates = get_predefined_prompt_templates()
+        self.prompt_templates: Dict[str, PromptTemplate] = {pt.name: pt for pt in prompt_templates}  # type: ignore
```

Comment:
@julian-risch I have to do it this way rather than directly specify the get_predefined_prompt_templates() as the default parameter. If we specify `prompt_templates: Optional[List[PromptTemplate]] = get_predefined_prompt_templates()` parameter, then the method gets executed at class load time! And that's precisely what we don't want!

Reply:
Irrelevant now, resolving